src/includes/service.h

```diff
@@ -314,6 +314,12 @@ class service_record
     uid_t socket_uid = -1;  // socket user id or -1
     gid_t socket_gid = -1;  // socket group id or -1
 
+    string logfile;      // path to the logfile for record service outputs
+    int logfile_perms = 0;      // logfile permissions("mode")
+    uid_t logfile_uid = -1;      // logfile user id or -1
+    uid_t logfile_uid_gid = -1;  // logfile primary group id or -1
```

Comment:
This (`logfile_uid_gid`) shouldn't be here - it is never set, never used.

Reply:
Also, none of these settings belongs in `service_record`. Logging is specific to process-based services, so these belong in `base_process_service`. (Adding `logfile` here duplicates the variable that is already defined there).

Reply:
So I need to remove code about these things in `run-child-proc.cc`? I think yes

Reply:
No! You need to move these variables to the `base_process_service` class (except logfile, which is already there, and logfile_uid_gid, which isn't needed at all).

Reply:
`logfile` is still here

Reply:
I saw that and removed that from `service.h`, this PR doesn't modify something in `service.h` now.